yarn-project/circuit-types/src/interfaces/aztec-node.ts

```diff
@@ -316,6 +316,13 @@ export interface AztecNode {
    **/
   simulatePublicCalls(tx: Tx): Promise<PublicSimulationOutput>;
 
+  /**
+   * Validates the correctness of the execution, namely that a transaction is valid if and
+   * only if the transaction can be added to a valid block at the current state.
+   * @param tx - The transaction to validate for correctness.
+   */
+  validateTx(tx: Tx): Promise<boolean>;
```

Comment:
I'd consider renaming this to e.g. `isTxValid` - 'validate' to me hints at some side effect potentially taking place